robottelo/performance/perf_candlepin.py

```diff
@@ -0,0 +1,116 @@
+"""Test utilities for writing Candlepin tests
+
+Part of functionalities of Candlepin are defined in this module
+and have utilities of single register by activation-key, single
+register and attach, single subscription deletion.
+
+"""
+import logging
+import time
+import requests
+
+from robottelo.common import conf, ssh
+from robottelo.common.helpers import get_server_credentials, get_server_url
+from urlparse import urljoin
+
+
+class CandlepinPerfTest(object):
+    """Measures performance of RH Satellite 6
+
+    Candlepin Subscription functionality
+
+    """
+
+    logger = logging.getLogger('robottelo')
+
+    # parameters for Candlepin Authentication
+    username = conf.properties['foreman.admin.username']
+    password = conf.properties['foreman.admin.password']
+    serverhost = conf.properties['main.server.hostname']
+
+    requests.packages.urllib3.disable_warnings()
+
+    @classmethod
+    def single_register_activation_key(cls, ak_name, default_org, vm):
+        """Subscribe VM to Satellite by Register + ActivationKey"""
+
+        # note: must create ssh keys for vm if running on local
+        result = ssh.command('subscription-manager clean', hostname=vm)
+        result = ssh.command(
+            'time -p subscription-manager register '
+            '--activationkey={0} --org={1}'
+            .format(ak_name, default_org), hostname=vm)
```

Comment:
Whenever reasonable, place each argument on its own line. In this case,
`hostname=vm` should be on its own line.

```python
        result = ssh.command(
            'time -p subscription-manager register --activationkey={0} '
            '--org={1}'.format(ak_name, default_org),
            hostname=vm,
        )
```